### ISSUE TITLE
depth_from_plane_equation: default eps=1e-8 rounds to 0 in float16, so a grazing ray returns inf

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4803]
`depth_from_plane_equation` guards the ray-plane dot product by replacing any `|denom| < eps` with `±eps` (`+eps` for an exact zero, since #4280). The default `eps = 1e-8` is below float16's smallest subnormal (`5.96e-08`), so in float16 it rounds to `0`. The guard then replaces a zero denominator with zero, and a grazing ray returns `inf`, even where the guarded answer `d / eps` is representable in float16.

## Repro

```python
import torch
from kornia.geometry.depth import depth_from_plane_equation

for dt in (torch.float32, torch.bfloat16, torch.float16):
    K = torch.tensor([[100.0, 0.0, 4.0], [0.0, 100.0, 3.0], [0.0, 0.0, 1.0]], dtype=dt)[None]
    px = torch.tensor([[[4.0, 3.0]]], dtype=dt)   # principal point: ray (0, 0, 1)
    n = torch.tensor([[0.0, 1.0, 0.0]], dtype=dt) # plane parallel to that ray
    d = torch.tensor([[1e-4]], dtype=dt)
    print(dt, depth_from_plane_equation(n, d, px, K).item(), depth_from_plane_equation(n, d, px, K, eps=1e-4).item())
```

```
torch.float32 10000.0 1.0
torch.bfloat16 9984.0 1.0
torch.float16 inf 1.0
```

Measured on `main` (`750c1611e`), torch 2.14.0, CPU.

## Expected

A grazing ray should return the large signed depth the guard promises in every supported dtype, not only in float32/float64/bfloat16. AGENTS.md describes this pattern: "A floor below the dtype's smallest subnormal is no guard either." Two fix directions:
- floor `eps` at the input dtype's smallest positive subnormal, as `normalize_quaternion` does after #4162;
- or reject an `eps` that rounds to zero in the input dtype.

Either fix should add a float16 pin whose guarded answer fits the dtype, such as the repro above. The existing grazing-ray pins in `tests/geometry/test_depth.py` skip float16 because their `2e8` depth overflows float16 whatever `eps` is.

The Convention block of `depth_from_plane_equation` states this as a wart linked here once #4790 merges.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
